src/rvi.c

```diff
@@ -640,11 +635,26 @@ int rviReadJsonConfig ( TRviHandle handle, const char * filename )
     return err;
 }
 
+/* This utility function returns a pointer to the first delim in a buffer, or
+ * NULL if delim does not appear in the buffer. IT DOES NOT COPY THE BUFFER: if
+ * needed, the partial buffer must be copied to a new buffer and the original
+ * buffer freed by the calling function.
+ */
+char *start_buf_at_char(char *buf, char delim)
+{
+    if (!buf) return NULL;
+    while (*buf != '\0' && *buf != delim && *buf++);
+    if (*buf != delim) {
+        buf = NULL;
+    }
+    return buf;
+}
+
 /* This utility function destructively reads from 'src' to populate '*json'
  * with a parsed json_t object. The remaining string (possibly 'src' exactly)
  * is copied into '*remainder'. The original 'src' must not be accessed again
  * by the calling function. */
-int rviReadJsonChunk( json_t **json, char *src, char **remainder )
+int rviReadJsonChunk( json_t **json, char *src, TRviRemote *remote )
```

Comment:
Not sure if remainder and remote are the same thing?  But there's a fair amount of rewrite here - would be nicer to put it in a separate commit?